### DISCUSSION TITLE
How to build FastDoom on Windows ?

### DISCUSSION THREAD

**Discussion by tigrouind** [viti95/FastDoom#205] [Q&A] [answered]
There is a list of instructions in the README, however it's not clear what to download exactly.
- [Open Watcom C version 2.0](https://github.com/open-watcom/open-watcom-v2/releases) : which version should be downloaded ? dos or win32 ?
- [NASM](https://www.nasm.us/pub/nasm/releasebuilds/) Same as above
- [DOS/32](https://sourceforge.net/projects/dos32a/) : should dos32a.exe be copied somewhere ?
- makeall.bat : it's now replaced by .sh file

**Comment by viti95** [comment, marked as answer]
Yep, I have to update the build instructions. 

All current develop is made on Linux or WSL2 (Windows). You need to install OpenWatcom v2, NASM, GNU Make and DOSBox-X.

Once everything is installed, these are the following steps to build:

1. Execute `source env.sh`
2. Build the executable:
- Single executable: `build.sh <executable> <parameters>` Where:
       "executable" is the name for the executable and "parameters" can be one of these:
       "-clean": cleans all generated OBJs, 
       "-stub": adds the DOS/32 stub, 
       "-debug": generates debugging exectable, needs dbgcfg.h to be adjusted to your needs
- All the executables (+ automatic DOS/32 stubs): `buildall.sh`
- Generate a full release zip: `package.sh`

For example, to build a fresh new executable:

```
./source env.sh
./build.sh fdoom.exe -clean -stub
```

**Comment by tigrouind** [comment]
Here is how to install everything under Ubuntu (can be run in Windows under with VirtualBox). 
Small (2GB) and ready to be used vdi disk images exists.

# DOSBox-X 
can be installed from App Center
# GNU Make
run `sudo apt install build-essential`
# OpenWatcom V2 : 
- open GitHub [release](https://github.com/open-watcom/open-watcom-v2/releases/tag/Current-build) page. 
- download [open-watcom-2_0-c-linux-x64](https://github.com/open-watcom/open-watcom-v2/releases/download/Current-build/open-watcom-2_0-c-linux-x64)
- mark file as executable (right click, properties)
- run `sudo ./open-watcom-2_0-c-linux-x64`
- use tab to navigate. Choose Next. Make sure "Full installation" is chosen.
# NASM 
- go to [NASM](https://www.nasm.us/) homepage. 
- Click on Downloads.
- Download [nasm-2.16.03.tar.gz](https://www.nasm.us/pub/nasm/releasebuilds/2.16.03/nasm-2.16.03.tar.gz) (or a later release).
- Extract archive
- run `./configure`
- run `make`
- run `sudo make install`
